pdns/recursordist/stable-bloom.hh

```diff
@@ -106,15 +106,29 @@ public:
     uint8_t k, p;
     uint32_t num_cells, bitstr_len;
     is.read((char*)&k, sizeof(k));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     is.read((char*)&num_cells, sizeof(num_cells));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     num_cells = ntohl(num_cells);
     is.read((char*)&p, sizeof(p));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     is.read((char*)&bitstr_len, sizeof(bitstr_len));
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
     bitstr_len = ntohl(bitstr_len);
-    char* bitcstr = new char[bitstr_len];
-    is.read((char*)bitcstr, bitstr_len);
-    std::string bitstr(bitcstr, bitstr_len);
-    delete[] bitcstr;
+    unique_ptr<char[]> bitcstr = make_unique<char[]>(bitstr_len);
+    is.read(bitcstr.get(), bitstr_len);
+    if (is.fail()) {
+      throw std::runtime_error("SBF: read failed (file too short?)");
+    }
+    std::string bitstr(bitcstr.get(), bitstr_len);
```

Comment:
Do we care about performance? If we do we could allocate and read into the string instead of using a temporary buffer.

Reply:
It's only at startup, it does take about a second per file on my system.

Reply:
I left this for now, if/when we change the on-disk format, we can make it fast.